frontend/taipy-gui/src/components/Taipy/TimeSelector.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+import React, { useState, useMemo, useCallback} from "react";
+import Box from "@mui/material/Box";
+import Tooltip from "@mui/material/Tooltip";
+import { ErrorBoundary } from "react-error-boundary";
+import { TimePicker } from '@mui/x-date-pickers/TimePicker';
+import { isValid } from "date-fns";
+import { MobileTimePicker } from '@mui/x-date-pickers/MobileTimePicker';
+import { getSuffixedClassNames, TaipyActiveProps, TaipyChangeProps, getCssSize } from "./utils";
+import { createSendUpdateAction } from "../../context/taipyReducers";
+import { useClassNames, useDispatch, useDynamicProperty, useModule } from "../../utils/hooks";
+import ErrorFallback from "../../utils/ErrorBoundary";
+import Field from "./Field";
+
+
+interface TimeSelectorProps extends TaipyActiveProps, TaipyChangeProps {
+  asClock?: boolean;
+  format?: string;
+  defaultDate?: string;
+  defaultTime?: string;
```

Comment:
default property is named value in factory.py

For date_selctor it is called [date](https://github.com/Avaiga/taipy/blob/8379661f1380a7722f1643b1252f62df55b5df8d/taipy/gui/_renderers/factory.py#L37)

Reply:
and value is initialized like 
`const [value, setValue] = useState(() => getDateTime(props.defaultDate, tz, withTime));`

And then every time props.date gets updated
```ts
    useEffect(() => {
        try {
            if (props.date !== undefined) {
                setValue(getDateTime(props.date, tz, withTime));
            }

            if (min !== undefined) {
                setStartProps((p) => getProps(p, true, getDateTime(min, tz, withTime), withTime));
            }

            if (max !== undefined) {
                setEndProps((p) => getProps(p, false, getDateTime(max, tz, withTime), withTime));
            }
        } catch (error) {
            console.error(error);
        }
    }, [props.date, tz, withTime, max, min]);
```

Reply:
Sorry I didn't figure it out at the first clue :smile: 
But now it's resolved I think